### PR TITLE
Xcode generation, Xlinker Xcc and Xswiftc options

### PR DESCRIPTION
**Pull request by goloveychuk**
Resolves https://bugs.swift.org/browse/SR-1169

### CODE REVIEW

File: Sources/Xcodeproj/generate().swift [hunk 1]
```diff
@@ -12,11 +12,13 @@ import PackageType
 import Utility
 import POSIX
 
-/** 
+public typealias OptionsType = (Xcc: [String], Xld: [String], Xswiftc: [String])
+
```
Comment (paulofaria) [inline Sources/Xcodeproj/generate().swift]: I think this could be just `Options`

**Comment by ddunbar** [comment]
Thanks for the PR! Can you please rebase your changes on to TOT so it is clear what exactly is new? Generally speaking a PR shouldn't have merge commits in it.

**Comment by goloveychuk** [comment]
@ddunbar done!

**Comment by mxcl** [comment]
LGTM

**Comment by mxcl** [comment]
@swift-ci Please test

**Comment by goloveychuk** [comment]
@mxcl I fixed. Don't have master swift to test

**Comment by ddunbar** [comment]
@swift-ci please test

**Comment by ddunbar** [comment]
@swift-ci please test

**Comment by ddunbar** [comment]
Tested this locally, previous failure was just the transient XCTest issue.

**Pull request merged**